bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/NibeGW/Debug.cpp

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Author: lassi.niemisto@gmail.com
+ *
+ */
+
+#include "Debug.h"
+
+#ifdef ENABLE_DEBUG
+
+char debugBuf[DEBUG_BUFFER_SIZE];
+
+#ifdef ENABLE_REMOTE_DEBUG
+  #if (CONN_MODE == CONN_MODE_ETH)
+    #include <Ethernet.h>
```

Comment:
This cause and issue with ProDino board (use different EthernetServer imp).

```
Debug.cpp:27:20: error: cannot declare variable 'telnet' to be of abstract type 'EthernetServer'
     EthernetServer telnet(23);
                    ^
In file included from /Users/pali/Desktop/lassi/NibeGW/Debug.cpp:25:0:
/Applications/Arduino.app/Contents/Java/libraries/Ethernet/src/Ethernet.h:253:7: note:   because the following virtual functions are pure within 'EthernetServer':
 class EthernetServer : public Server {
       ^
In file included from /Users/pali/Library/Arduino15/packages/esp32/hardware/esp32/1.0.6/cores/esp32/Arduino.h:152:0,
                 from /Users/pali/Desktop/lassi/NibeGW/Config.h:19,
                 from /Users/pali/Desktop/lassi/NibeGW/Debug.h:19,
                 from /Users/pali/Desktop/lassi/NibeGW/Debug.cpp:17:
/Users/pali/Library/Arduino15/packages/esp32/hardware/esp32/1.0.6/cores/esp32/Server.h:28:18: note: 	virtual void Server::begin(uint16_t)
     virtual void begin(uint16_t port=0) =0;

```

If I replaced it with #include "KMPProDinoESP32.h", then next issue is with 

```
/Users/pali/Desktop/lassi/NibeGW/Debug.cpp: In function 'bool getDebugInput(char*)':
Debug.cpp:58:14: error: 'class EthernetServer' has no member named 'hasClient'
   if (telnet.hasClient()) {
```

as KMPProDinoESP32 EthernetServer doesn't implement hasClient function.

Reply:
Thanks, I have now fixed the compilation for Prodino again